ProcessMaker/processmaker
# Fix: Reduce Database Bloat in Guided Templates

## Issue & Reproduction Steps
This PR resolves an issue with Guided Templates causing database bloat whenever a guided template sync command is run multiple times. The problem stemmed from the creation of new version records for all guided template assets each time a guided template sync command was executed. This resulted in unintended database bloat and increased spin-up time for servers required by the PLG team.

## Solution
To address this, the solution involves a hashing mechanism that only imports the helper process or process template when the hash has changed.  Unnecessary database bloat is prevented, leading to improved database performance and reduced resource consumption.

## How to Test

1. Ensure access to an environment database.
2. Run a guided template sync command multiple times.
3. Check the asset versions table (e.g., screen_versions, script_versions) in the database.
4. Verify that no new version records are created for guided template assets after running the guided template.
5. Repeat the test with multiple guided templates to confirm consistent behavior.
6. Compare database size and performance metrics before and after implementing the fix to observe the reduction in database bloat and improved performance.

## Related Tickets & Packages
- [FOUR-15105](https://processmaker.atlassian.net/browse/FOUR-15105)

ci:next
ci:GUIDED_TEMPLATE_BRANCH=observation/FOUR-15105


## Code Review Checklist


[FOUR-15105]: https://processmaker.atlassian.net/browse/FOUR-15105?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
